sycl/include/syclcompat/util.hpp

```diff
@@ -161,6 +161,11 @@ inline double cast_ints_to_double(int high32, int low32) {
 template <typename T> inline T reverse_bits(T a) {
   static_assert(std::is_unsigned<T>::value && std::is_integral<T>::value,
                 "unsigned integer required");
+#if defined(__NVPTX__)
```

Comment:
Could you `if constexpr(std::is_same_v<T, unsigned>` here? Theoretically this API accepts e.g. `unsigned short`, `unsigned char` and I guess the PTX wouldn't work for that.

Of course, the ptx for those cases could be defined too, so that'd be an alternative.

Reply:
Thanks for pointing this out. I've dealt with this using `sizeof`